apps/trading/client-pages/markets/closed.tsx

```diff
@@ -64,15 +64,18 @@ export const Closed = () => {
     const instrument = market.tradableInstrument.instrument;
 
     const spec =
+      'dataSourceSpecForSettlementData' in instrument.product &&
       instrument.product.dataSourceSpecForSettlementData.data.sourceType
         .__typename === 'DataSourceDefinitionExternal'
         ? instrument.product.dataSourceSpecForSettlementData.data.sourceType
             .sourceType
         : undefined;
-    const filters = spec?.filters || [];
+    const filters = (spec && 'filters' in spec && spec.filters) || [];
 
     const settlementDataSpecBinding =
-      instrument.product.dataSourceSpecBinding.settlementDataProperty;
+      'dataSourceSpecBinding' in instrument.product
```

Comment:
I'd refactor the Row type, so that we have a specBinding field which is a union based on product type